autodispose-android-archcomponents/src/main/java/com/uber/autodispose/android/lifecycle/AndroidLifecycleScopeProvider.java

```diff
@@ -54,41 +54,84 @@
         }
       };
 
+  private final Function<Lifecycle.Event, Lifecycle.Event> correspondingEvents;
+
   /**
    * Creates a {@link AndroidLifecycleScopeProvider} for Android LifecycleOwners.
    *
-   * @param owner the owner to scope for
+   * @param owner the owner to scope for.
    * @return a {@link AndroidLifecycleScopeProvider} against this owner.
    */
   public static AndroidLifecycleScopeProvider from(LifecycleOwner owner) {
     return from(owner.getLifecycle());
   }
 
   /**
+   * Creates a {@link AndroidLifecycleScopeProvider} for Android LifecycleOwners.
+   *
+   * @param owner the owner to scope for.
+   * @param untilEvent the event until the scope is valid.
+   * @return a {@link AndroidLifecycleScopeProvider} against this owner.
+   */
+  public static AndroidLifecycleScopeProvider from(LifecycleOwner owner,
+                                                   Lifecycle.Event untilEvent) {
+    return from(owner.getLifecycle(), untilEvent);
+  }
+
+  /**
    * Creates a {@link AndroidLifecycleScopeProvider} for Android Lifecycles.
    *
-   * @param lifecycle the lifecycle to scope for
+   * @param lifecycle the lifecycle to scope for.
    * @return a {@link AndroidLifecycleScopeProvider} against this lifecycle.
    */
   public static AndroidLifecycleScopeProvider from(Lifecycle lifecycle) {
     return new AndroidLifecycleScopeProvider(lifecycle);
   }
 
+  /**
+   * Creates a {@link AndroidLifecycleScopeProvider} for Android Lifecycles.
+   *
+   * @param lifecycle the lifecycle to scope for.
+   * @param untilEvent the event until the scope is valid.
+   * @return a {@link AndroidLifecycleScopeProvider} against this lifecycle.
+   */
+  public static AndroidLifecycleScopeProvider from(Lifecycle lifecycle, Lifecycle.Event untilEvent) {
+    return new AndroidLifecycleScopeProvider(lifecycle, untilEvent);
+  }
+
   private final LifecycleEventsObservable lifecycleObservable;
 
   private AndroidLifecycleScopeProvider(Lifecycle lifecycle) {
     this.lifecycleObservable = new LifecycleEventsObservable(lifecycle);
+    this.correspondingEvents = DEFAULT_CORRESPONDING_EVENTS;
+  }
+
+  private AndroidLifecycleScopeProvider(Lifecycle lifecycle, Lifecycle.Event untilEvent) {
+    this.lifecycleObservable = new LifecycleEventsObservable(lifecycle);
+    this.correspondingEvents = new UntilEventFunction(untilEvent);
   }
 
   @Override public Observable<Lifecycle.Event> lifecycle() {
     return lifecycleObservable;
   }
 
   @Override public Function<Lifecycle.Event, Lifecycle.Event> correspondingEvents() {
-    return CORRESPONDING_EVENTS;
+    return correspondingEvents;
   }
 
   @Override public Lifecycle.Event peekLifecycle() {
     return lifecycleObservable.getValue();
   }
+
+  private static class UntilEventFunction implements Function<Lifecycle.Event, Lifecycle.Event> {
```

Comment:
Good idea. I'd initially thought about doing it as a filter, but with this you hook in nicely to the api

Reply:
Yeah I started trying that but could not get it to work, so came up with this approach instead.